manim/utils/simple_functions.py

```diff
@@ -54,28 +55,16 @@ def binary_search(
         >>> binary_search(lambda x: x**2 + 3*x + 1, 71, 0, 5) is None
         True
     """
-    lh = lower_bound
-    rh = upper_bound
-    mh = np.mean(np.array([lh, rh]))
-    while abs(rh - lh) > tolerance:
-        mh = np.mean(np.array([lh, rh]))
-        lx, mx, rx = (function(h) for h in (lh, mh, rh))
-        if lx == target:
-            return lh
-        if rx == target:
-            return rh
-
-        if lx <= target <= rx:
-            if mx > target:
-                rh = mh
-            else:
-                lh = mh
-        elif lx > target > rx:
-            lh, rh = rh, lh
-        else:
-            return None
-
-    return mh
+    inputs = [
+        lower_bound + i * (upper_bound - lower_bound) / 10000 for i in range(10001)
```

Comment:
10000 seems like a rather arbitrary value. Where does it come from?

Reply:
Used in this code to generate a list of 10,000 evenly spaced values between lower_bound and upper_bound with list comprehension

Reply:
Arbitrary fr
I'm surfing the net to find ways to implement the function better using the bisect module